code/script.js

```diff
@@ -0,0 +1,147 @@
+      const sortBtn = document.getElementById('sort')
+      sortBtn.addEventListener('change', () => sort(sortBtn.value, forkedRepos, commitslength))
+
+			drawProjects(forkedRepos);	
+			drawChart(forkedRepos.length)
+		})
+}
+
+//sorting the projects by name, last updated or no. of commits
+const sort = (value, repos, mycommits) => {
+  if (value === "name") {
+    repos.sort((a, b) => {
+      return (a.name) > (b.name);
+    });
+    drawProjects(repos)
+  } else if (value === "updated") {
+    repos.sort((a, b) => {
+      return new Date(b.pushed_at) - new Date(a.pushed_at);
+    });
+      drawProjects(repos)
+  } /* else if (value === "commits") {
+    console.log('Commits: ', mycommits)
+    console.log('Commitslänge: ', mycommits.length)
+    mycommits.sort((a, b) => {
+      return (a.mycommits.length) > (b.mycommits.length);
+    });
+    drawProjects(repos) 
+  }*/
+}
+
+const drawProjects = (forkedRepositories) => {
+	document.getElementById('projects-section').innerHTML = `
+	`
+  console.log('From drawProjects()', forkedRepositories)
+	forkedRepositories.forEach((repo) => {
+		document.getElementById('projects-section').innerHTML += `
+			<div class="projects-div" id="projects">
+				<a href="${repo.html_url}">${repo.name}</a>
+				<p>default branch: ${repo.default_branch}</p>
+				<p>Last push: ${new Date(repo.pushed_at).toLocaleDateString('en-GB')}</p>
+				<p class="noOfCommits" id="commit-${repo.name}">Commits: 0</p>
+        		<ul id="commitMessages-${repo.name}"></ul>
+			</div>	
+		`;
+
+		getCommits(forkedRepositories, repo.name);
+	});
+
+	forkedRepositories.forEach((repo) => {
+		document
+			.getElementById(`commit-${repo.name}`)
+			.addEventListener('click', () => {
+				document
+					.getElementById(`commitMessages-${repo.name}`)
+					.classList.toggle('active');
+			});
+	});
+};
+
+const getCommits = (filteredArray, myRepoName) => {
+	console.log('Reponame: ', myRepoName)
+	//First make a new array with all the needed APIs (found under commit_urls in forkedRepos)
+	const allCommitUrls = filteredArray.map(repo => repo.commits_url)
+	
+	allCommitUrls.forEach(commitAPI => {
+		//the URLs end with {/sha}, therefore the last 6 chars need to be sliced
+		commitAPI = commitAPI.slice(0, -6)
+
+		//now the URLs can be passed on to get data about number and content of commits 
+		fetch(commitAPI, /* options */)
+			.then((res) => res.json())
+			.then((data) => {
+				const authorCommits = data.filter(commits => commits.author.login === 'nehrwein' && commits.url.includes(myRepoName))
+				if (authorCommits.length > 0) {
+					commitslength.push(authorCommits)
+					document.getElementById(`commit-${myRepoName}`).innerHTML = `
+						Commits: ${authorCommits.length}<i class="fas fa-bars"></i>
+					`
+					authorCommits.forEach(element => {
+						document.getElementById(`commitMessages-${myRepoName}`).innerHTML += `
+						<li>${element.commit.message}</li>
+						`
+					})
+				} 
+			})		
+	});
+}
+
+
+
+getRepos();
+
```

Comment:
I really love our code Birgit and it was so interesting on Friday when you showed it to the group. I think you made an awesome site!!! The implementation of an accordion is just the next level of the project! everything seems to work fine the only trouble I have is whit the sorting function (the name sorting is not working for me).  That's not a mandatory feature so if you don't have the time not bother to fix it! You and I did solve some functions in different ways and it is really fun to see our solutions, the code is easy to read and understand!

 [YES ]  Am I able to understand the code easily?
 [NO ]  Does the code have lots of duplicates? Could it be tidied up?
 [ works ]  Does it work or does anything look broken?
 [Yes and above! ]  Does it follow the general and blue requirements?

Reply:
Hej Jessica,
thank you so much for your kind review! 👍 
And thanks for pointing out, that the sorting isn't working. I will have a look at it again. 🙂